clash-prelude/src/Clash/Class/BitPack/Internal.hs

```diff
@@ -478,7 +479,15 @@ instance ( BitPack a
          ) => BitPack (Either a b)
 
 instance BitPack a => BitPack (Maybe a)
-instance BitPack (Proxy a) => BitPack (Proxy a)
+
+-- |
+-- >>> pack (Proxy @())
+-- 0
+-- >>> pack (Proxy @5)
+-- 0
+-- >>> pack (Proxy @Bool)
+-- 0
+instance BitPack (Proxy a)
```

Comment:
Unless the reader is very familiar with the details of the `Show BitVector` instance, it might not be clear from the example above that that `BitSize (Proxy a) ~ 0`.
(I missed this point myself, reading it the first time.)

Maybe we should explicit like document that.

Reply:
Yes, the 0 is pretty confusing at first! I completely agree we should document it explicitly as a 0-sized construct.